specification/cognitiveservices/OpenAI.Inference/models/completions/extensions_onyourdata_authentication.tsp

```diff
@@ -0,0 +1,89 @@
+import "@typespec/versioning";
+
+namespace Azure.OpenAI;
+
+using TypeSpec.Versioning;
+
+@doc("""
+  Authentication types supported by with Azure OpenAI on your data.
+""")
+@added(ServiceApiVersions.v2023_08_01_Preview)
+enum OnYourDataAuthenticationType {
+  @doc("API key.")
+  apiKey: "APIKey",
+
+  @doc("Connection string.")
+  connectionString: "ConnectionString",
+
+  @doc("Key and keyId pair.")
+  keyAndKeyId: "KeyAndKeyId",
+
+  @doc("System-assigned managed identity.")
+  systemAssignedManagedIdentity: "SystemAssignedManagedIdentity",
+
+  @doc("User-assigned managed identity.")
+  userAssignedManagedIdentity: "UserAssignedManagedIdentity",
+}
+
+@discriminator("type")
+@doc("The authentication options for Azure OpenAI on your data.")
+@added(ServiceApiVersions.v2023_08_01_Preview)
+model OnYourDataAuthenticationOptions {
+  @doc("The Authentication type.")
+  type: OnYourDataAuthenticationType;
+}
+
+@doc("The authentication option using API key.")
```

Comment:
nit: A more complete description might read better for this one and the other derived types, e.g.: "The authentication options for the Azure OpenAI "on your data" extension when using an API key."

Reply:
Agreed, I'll edit these across the board.